tests/unit/wait.tcl

```diff
@@ -140,6 +140,32 @@ tags {"wait aof network external:skip"} {
             assert_error {ERR WAITAOF cannot be used when numlocal is set but appendonly is disabled.} {$master waitaof 1 0 0}
         }
 
+        test {WAITAOF local if AOFRW was postponed} {
+            r config set appendfsync everysec
+
+            # turn off AOF
+            r config set appendonly no
+
+            # create an RDB child that takes a lot of time to run
+            r set x y
+            r config set rdb-key-save-delay 100000000  ;# 100 seconds
+            r bgsave
+
+            # turn on AOF
+            r config set appendonly yes
+            assert_equal [s aof_rewrite_scheduled] 1
+
+            # create a write command (to increment master_repl_offset)
+            r set x y
+
+            # reset save_delay and kill RDB child
+            r config set rdb-key-save-delay 0
+            catch {exec kill -9 [get_child_pid 0]}
+
+            # wait for AOF
+            assert_equal [r waitaof 1 0 1000] {1 0}
```

Comment:
do we wanna also wait for the rewrite to start (or even complete) in that test?

Reply:
WAITAOF will implicitly wait for the fork to start and finish

Reply:
generally i'd like to remove timing issues from this test, so that the order of things is always the same.
i realize you're right since it's an new AOF (not a normal rewrite), WAITAOF will wait for it to finish.
still i'd like to see the test refers to it. maybe we can assert that after the WAITAOF returns the rewrite is completed already? (performed and no longer pending).

Reply:
do you mean just adding `assert_equal [s aof_rewrite_scheduled] 0` after waitaof?

Reply:
would 1000 be a bit short if some machines are very slow?

Reply:
> do you mean just adding `assert_equal [s aof_rewrite_scheduled] 0` after waitaof?

yes, and also `aof_rewrite_in_progress`.
if we're paranoid, we can also check that `aof_rewrites` was incremented.



> would 1000 be a bit short if some machines are very slow?

we can even use an infinite timeout? if there's ever a bug.. i don't mind the test suite to hang.
on the other hand, 10 seconds could be ok too.

Reply:
@oranagra i think @enjoy-binbin that WAITAOF will timeout if AOFRW takes more than 1 second

will change to 10